Formula/p/postgresql@17.rb

```diff
@@ -0,0 +1,189 @@
+class PostgresqlAT17 < Formula
+  desc "Object-relational database system"
+  homepage "https://www.postgresql.org/"
+  url "https://ftp.postgresql.org/pub/source/v17.0/postgresql-17.0.tar.bz2"
+  sha256 "7e276131c0fdd6b62588dbad9b3bb24b8c3498d5009328dba59af16e819109de"
+  license "PostgreSQL"
+
+  livecheck do
+    url "https://ftp.postgresql.org/pub/source/"
+    regex(%r{href=["']?v?(17(?:\.\d+)+)/?["' >]}i)
+  end
+
+  bottle do
+    sha256 arm64_sequoia: "7345b4a703b655b5ef7bcb37a3dbc90b5406e4d06e7c4b8d30f8bd43f8499974"
+    sha256 arm64_sonoma:  "34876e4d4b26909d79f757d60f7cb34c6e2531c03120f103f5063a6cf62facd3"
+    sha256 arm64_ventura: "f2facfda470d494aea051eb64ba54c60171e5866fa032ef7c3fba3b488a69f13"
+    sha256 sonoma:        "a5258e4c8218b8ba6d91bb368581548042dc618e07a788e19d0fdf6b7297d7fd"
+    sha256 ventura:       "e25fa2f7a5dc496172f262913071f1a85ce03711b1edfd41a8e9f4a38d2c0b2e"
+    sha256 x86_64_linux:  "3a5e1daf6f1cef652cb39b869e0d8490c3bf29fcadc1869fbe7c92a7f37d6810"
+  end
+
+  keg_only :versioned_formula
+
+  # https://www.postgresql.org/support/versioning/
+  deprecate! date: "2029-11-08", because: :unsupported
+
+  depends_on "docbook" => :build
+  depends_on "docbook-xsl" => :build
+  depends_on "gettext" => :build
+  depends_on "pkg-config" => :build
+  depends_on "icu4c"
+  # GSSAPI provided by Kerberos.framework crashes when forked.
+  # See https://github.com/Homebrew/homebrew-core/issues/47494.
+  depends_on "krb5"
+  depends_on "lz4"
+  depends_on "openssl@3"
+  depends_on "readline"
+  depends_on "zstd"
+
+  uses_from_macos "bison" => :build
+  uses_from_macos "flex" => :build
+  uses_from_macos "libxml2"
+  uses_from_macos "libxslt"
+  uses_from_macos "openldap"
+  uses_from_macos "perl"
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on "gettext"
+  end
+
+  on_linux do
+    depends_on "linux-pam"
+    depends_on "util-linux"
+  end
+
+  def install
+    # Modify Makefile to link macOS binaries using Cellar path. Otherwise, binaries are linked
+    # using #{HOMEBREW_PREFIX}/lib path set during ./configure, which will cause audit failures
+    # for broken linkage as the paths are not created until post-install step.
+    inreplace "src/Makefile.shlib", "-install_name '$(libdir)/", "-install_name '#{lib}/postgresql/"
```

Comment:
Looking at RPATHs, following may work:
```suggestion
    inreplace "src/Makefile.shlib", "-install_name '$(libdir)/", "-install_name '@rpath/"
```

With multiple `-rpath` passed in (I think 3 are needed for `@loader_path`, `@loader_path/../lib/postgresql`, and `@loader_path/../../../..`).

Can experiment some more in other PR.